### PR TITLE
Refactor/part 1

### PR DESCRIPTION
**Pull request by catlynbowles**
What does this PR do?
- Add `deep.equals` to testing of arrays in test files
- Move sample data set (for hydration and sleep) from test files into their own files
- Refactored two functions that obtain average sleep hours and quality info into 1 function
- Create a new function that reformats dates in any weekly datasets

Where should your reviewer start?
- Hydration.js line 40
- Sleep.js line 13
- scripts.js line 58
- Hydration-test.js line 3
- Sleep-test.js line 3
- UserRepository-test.js line 83
- Hydration-data.js line 1
- Sleep-data.js line 1

What (if any) features are you implementing?
- None

What (if anything) did you refactor?
- We focused on refactoring, as listed above. 

Were there any issues that arose?
- N/A

Is there anything that you need from your teammate?
- N/A

Any other comments, questions, or concerns?
- Closed Issues: #61 #73 #72 #60

### CODE REVIEW

File: test/UserRepository-test.js [hunk 1]
```diff
@@ -80,7 +80,8 @@ describe('User Repository', () => {
   });
 
   it('should be able to take in user data', () => {
-    expect(userRepository1.data).to.equal(testUserData);
+    expect(userRepository1.data).to.be.an('array');
```
Comment (irmakerdem) [inline test/UserRepository-test.js:83]: We should test for data type in each it block so we never forget to do a deep equals as a reminder (even though it passes the tests already) 🤡

File: src/scripts.js [hunk 2]
```diff
@@ -99,46 +109,44 @@ function displaySleepInfo(newUser) {
...
-  var hydrationObjectsDom = hydrationWeekly.map(oneDate => {
-    let myArray = oneDate.split(":");
-    let interp;
-    let date = myArray[0];
-    let [year, month, day] = date.split('/');
-    let result = [month, day, year].join('/');
-    interp = "Date:  " + result + '\xa0\xa0\xa0\xa0\xa0\xa0' + "Ounces:  " + myArray[1];
-    return interp;
-  });
+  var hydrationObjectsDom = good(hydrationWeekly, 'Ounces: ')
```
Comment (irmakerdem) [inline src/scripts.js:149]: good function 🫣 name needs work 🤪

File: src/Sleep.js [hunk 3]
```diff
@@ -10,23 +10,32 @@ class Sleep {
...
+  //  }, 0);
+  //  let aveHours = totalHours / sleepData.length;
+  //  return Number(aveHours.toFixed(1));
+  // }
 
-  acquireAvgSleepQualityPerDay(sleepData) {
-    let totalQuality = sleepData.reduce((total, day) => {
-      total += day.sleepQuality;
-      return total;
+  acquireAverageMetricPerDay(measurements) {
```
Comment (irmakerdem) [inline src/Sleep.js:23]: Love this name and functionality!! 🥳

**Comment by ASands17** [comment]
Awesome job team! We really dried up our code with this one! 🧽  🧽  🧽

**Pull request merged**